python/ray/data/tests/test_dataset_map.py

```diff
@@ -0,0 +1,920 @@
+import itertools
```

Comment:
Maybe a file level string to explain what this is covering (as what PR description says)?

Reply:
Yes make sense. I am thinking of adding a readme in test directory, to explain each file. I want to unbreak CI as soon as possible, and avoid merge conflict with other PR. Can we do it as followup?